src/content/docs/Generic Programming/compiletime.md

```diff
@@ -200,13 +196,19 @@ macro long @fib(long $n)
 }
 ```
 
-It is important to remember that if we had replaced `$n` with `n` the compiler would have complained. `n <= 1`
-is not be considered to be a constant expression, even if the actual argument to the macro was a constant.
-This limitation is deliberate, to offer control over what is compiled out and what isn't.
+It is important to remember that if we had replaced `$n` with `n` the compiler would have complained. `n <= 1` is not considered to be a constant expression, even if the actual argument to the macro was a constant. This limitation is deliberate, to offer control over what is compiled out and what isn't.
+
+:::note
+The act of code generation itself will still complete in compile time regardless of whether a macro is run time dependent or not. In that sense all macros could be said to "complete in compile time" when actually understood rigorously and clearly. As such, compile time code generation should not be confused with the run time effects of such generated code. 
```

Comment:
I found this note a bit difficult to read, could you shorten it down a bit perhaps?